presto-cli/src/main/java/io/prestosql/cli/ClientOptions.java

```diff
@@ -366,4 +379,66 @@ public boolean equals(Object obj)
                     Objects.equals(this.value, other.value);
         }
     }
+
+    public static final class ClientExtraCredentials
+    {
+        private final String name;
+        private final String value;
+
+        public ClientExtraCredentials(String extraCredentials)
+        {
+            List<String> nameValue = NAME_VALUE_SPLITTER.splitToList(extraCredentials);
+            checkArgument(nameValue.size() == 2, "Extra credential: %s", extraCredentials);
+
+            this.name = nameValue.get(0);
+            this.value = nameValue.get(1);
+            checkArgument(!name.isEmpty(), "Credential name is empty");
+            checkArgument(!value.isEmpty(), "Credential value is empty");
+            checkArgument(PRINTABLE_ASCII.matchesAllOf(name), "Credential name contains spaces or is not US_ASCII: %s", name);
+            checkArgument(name.indexOf('=') < 0, "Credential name must not contain '=': %s", name);
+            checkArgument(PRINTABLE_ASCII.matchesAllOf(value), "Credential value contains space or is not US_ASCII: %s", name);
+        }
+
+        @VisibleForTesting
```

Comment:
remove the annotation

Reply:
Agreed, it's part of the public interface for the class (needed to construct it manually and will be called by Airline framework)